app/models/search.rb

```diff
@@ -15,19 +16,34 @@ class Search
   validate :dont_search_the_past
 
   def date_value
-    begin
-      Date.strptime(self.date, '%d/%m/%Y')
-    rescue
-      Date.today
-    end
+    # dans le @search : '%d/%m/%Y'
+    # date en texte de type : "vendredi 13 mai 2011" : 4 chiffres -> l'année ; 1 ou 2 chiffres -> le jour ; le mois en lettres est dans I18n.t('date.month_names'), .index("mois en lettres") donne le mois en chiffres
+    #begin
+    #Date.strptime(self.date, '%d/%m/%Y')
+    the_match_object = /([A-Za-z]+) ([0-9]{1,2}) ([A-Za-z]+) ([0-9]{4})/.match(self.date)
```

Comment:
Might use `self.date.split` if you are sure of the format.
Else guard regex if some element does not exists.

In the long term we better fix I18n to return French date and so parse it see below.

```
I18n.locale
=> :fr
irb(main):002:0> Date.today.strftime('%A %d %B %Y')
# => "Monday 17 July 2017"
# Wrong should have in French
Date.strptime("vendredi 13 mai 2011", '%A %d %B %Y')
# So we could might parse it easily
```